File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/HeraldKali.java
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper;
import java.util.LinkedHashMap;

/* JADX INFO: loaded from: classes3.dex */
public class HeraldKali extends Enemy {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public int calculateManaRegen() {
        return 10;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMaxDamage() {
        return 305;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMinDamage() {
        return 280;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isMagic() {
        return true;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isRanged() {
        return true;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected void configureStatistics() {
        this.baseMaxHp = 4500;
        this.baseConstitution = 500;
        this.baseIntelligence = 500;
        this.baseDexterity = 500;
        this.baseDefense = 0;
        this.baseMagicDefense = 0;
        this.alwaysHits = true;
        this.immunityToStatus = 0.5d;
        this.currentMana = 100;
        this.imageId = R.drawable.unit_herald_kali;
        this.idName = R.string.enemy_herald_kali_name;
        this.idDescription = R.string.enemy_herald_kali_description;
        this.passiveSkill = Skills.PASSIVE_CLAIRVOYANCE;
        this.activeSkill = Skills.ACTIVE_SACRIFICE;
        this.rarity = 1;
        this.expGiven = 9600;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    public LinkedHashMap<ItemWrapper, Integer> listDrops(int i) {
        LinkedHashMap<ItemWrapper, Integer> linkedHashMap = new LinkedHashMap<>();
        linkedHashMap.put(ItemWrapper.getInstance("SerpentStaff", 1), 1000);
        return linkedHashMap;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/HeraldKali.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper
import java.util.LinkedHashMap

class HeraldKali : Enemy() {
    override fun getMaxDamage(): Int = 305
    override fun getMinDamage(): Int = 280
    override fun isMagic(): Boolean = true
    override fun isRanged(): Boolean = true

    override fun configureStatistics() {
        baseMaxHp = 4500
        baseConstitution = 500
        baseIntelligence = 500
        baseDexterity = 500
        baseDefense = 0
        baseMagicDefense = 0
        alwaysHits = true
        immunityToStatus = 0.5
        currentMana = 100
        imageId = R.drawable.unit_herald_kali
        idName = R.string.enemy_herald_kali_name
        idDescription = R.string.enemy_herald_kali_description
        passiveSkill = Skills.PASSIVE_CLAIRVOYANCE
        activeSkill = Skills.ACTIVE_SACRIFICE
        rarity = 1
        expGiven = 9600
    }

    override fun listDrops(i: Int): LinkedHashMap<ItemWrapper, Int> {
        val linkedHashMap = LinkedHashMap<ItemWrapper, Int>()
        linkedHashMap.put(ItemWrapper.getInstance("SerpentStaff", 1), 1000)
        return linkedHashMap
    }
    override fun calculateManaRegen(): Int = 10
}
